addons/web/static/src/core/utils/patch.js

```diff
@@ -25,135 +63,65 @@ const patchMap = new WeakMap();
+            Object.defineProperty(description.skeleton, key, oldProperty);
         }
 
-        if (prevDesc) {
-            const patchedFnName = `${k} (patch ${patchName})`;
-
-            if (prevDesc.value && typeof newDesc.value === "function") {
-                newDesc = { ...prevDesc, value: newDesc.value };
-                makeIntermediateFunction("value", prevDesc, newDesc, patchedFnName);
-            }
-            if ((newDesc.get || newDesc.set) && (prevDesc.get || prevDesc.set)) {
-                // get and set are defined together. If they are both defined
-                // in the previous descriptor but only one in the new descriptor
-                // then the other will be undefined so we need to apply the
-                // previous descriptor in the new one.
-                newDesc = {
-                    ...prevDesc,
-                    get: newDesc.get || prevDesc.get,
-                    set: newDesc.set || prevDesc.set,
-                };
-                if (prevDesc.get && typeof newDesc.get === "function") {
-                    makeIntermediateFunction("get", prevDesc, newDesc, patchedFnName);
-                }
-                if (prevDesc.set && typeof newDesc.set === "function") {
-                    makeIntermediateFunction("set", prevDesc, newDesc, patchedFnName);
-                }
-            }
+        if (!description.originalProperties.has(key)) {
+            // Keep a trace of original property (prop before first patch), useful for unpatching.
+            description.originalProperties.set(key, oldProperty);
         }
 
-        Object.defineProperty(obj, k, newDesc);
-    }
-
-    function makeIntermediateFunction(key, prevDesc, newDesc, patchedFnName) {
-        const _superFn = prevDesc[key];
-        const patchFn = newDesc[key];
-        if (pure) {
-            newDesc[key] = patchFn;
-        } else {
-            newDesc[key] = {
-                [patchedFnName](...args) {
-                    let prevSuper;
-                    if (this) {
-                        prevSuper = this._super;
-                        Object.defineProperty(this, "_super", {
-                            value: _superFn.bind(this),
-                            configurable: true,
-                            writable: true,
-                        });
-                    }
-                    const result = patchFn.call(this, ...args);
-                    if (this) {
-                        Object.defineProperty(this, "_super", {
-                            value: prevSuper,
-                            configurable: true,
-                            writable: true,
-                        });
-                    }
-                    return result;
-                },
-            }[patchedFnName];
+        // A property is enumerable on POJO ({ prop: 1 }) but not on classes (class A {}).
+        newProperty.enumerable = !isClass(reference) && !isClassPrototype(reference);
```

Comment:
Not sure this is what we want, maybe we just want to make all getters non-enumerable since this is what we're having issues with?